ws/api/requests.go

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Mainflux
+// SPDX-Licence-Identifier: Apache-2.0
+
+package api
+
+import "github.com/gorilla/websocket"
+
+type connReq struct {
+	thingKey string
+	chanID   string
+	subtopic string
+	conn     *websocket.Conn
```

Comment:
I would suggest using this like a pure DTO and not adding any logic here. You can leave this logic to the Client (because the client operates on the adapter level).